python/pylibcugraph/pylibcugraph/random.pyx

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+from pylibcugraph._cugraph_c.random cimport (
+    cugraph_rng_state_create,
+    cugraph_rng_state_free,
+    cugraph_rng_state_t,
+)
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle
+)
+
+import time
+import os
+import socket
+
+def generate_default_seed():
+    h = hash(
+            (
+                socket.gethostname(),
+                os.getpid(),
+                time.perf_counter_ns()
+            )
+        )
+    
+    # reinterpret as unsigned
+    return h & (2**64 - 1)
+
+class global_random_instance_wrapper:
+    global_random_instance = None
+
+cdef class CuGraphRandomState:    
```

Comment:
I'm a little confused by how this is supposed to be used. The `__cinit__` implies that that I can have multiple instances if I wanted to, but all methods are classmethods which refer to a global.  Perhaps some docstrings would help?

Reply:
multiple instances are ok, see docstring